src/components/ui/CopyLink/CopyLink.tsx

```diff
@@ -0,0 +1,75 @@
+import { AnimatePresence, motion } from 'framer-motion';
+import { useEffect, useState } from 'react';
+
+export const CopyLink = ({ link }: { link: string }) => {
+  const [copied, setCopied] = useState(false);
+
+  // Reset copied state after 1.5 seconds to allow user to copy again
+  useEffect(() => {
+    if (copied) {
+      const timer = setTimeout(() => {
+        setCopied(false);
+      }, 1500);
+      return () => clearTimeout(timer);
+    }
+  }, [copied]);
+
+  const copyLink = () => {
+    navigator.clipboard.writeText(link);
+    setCopied(true);
+  };
+
+  return (
+    <div className="flex items-center h-10 rounded-full justify-between px-2 pr-4 shadow-[0_0_0.5rem_rgba(4,41,49,0.15)] text-[0.813rem] relative">
+      <div className="whitespace-nowrap overflow-x-clip">{link}</div>
+      <div className="w-14 h-6 bg-gradient-to-l from-white right-[5.5rem] absolute"></div>
+      <AnimatePresence>
+        {!copied ? (
+          <motion.div
+            key="copy-link"
+            onClick={copyLink}
+            initial={{ opacity: 1 }}
+            animate={{ opacity: 1 }}
+            exit={{ opacity: 0 }}
+            className="flex items-center gap-[0.281rem] rounded-full py-1 px-2 bg-white hover:bg-[#DEE5ED] transition duration-300 cursor-pointer z-0 absolute right-0 mr-2"
+          >
+            <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+              <path
+                d="M6.24269 12.3333L5.62135 12.9546C5.05874 13.5172 4.29567 13.8333 3.50002 13.8333C2.70436 13.8333 1.9413 13.5172 1.37869 12.9546C0.816073 12.392 0.5 11.6289 0.5 10.8333C0.5 10.0376 0.816073 9.27455 1.37869 8.71194L4.56069 5.52927C5.09417 4.99459 5.8097 4.68081 6.56441 4.6506C7.31911 4.62039 8.05744 4.87597 8.63195 5.3663C9.20647 5.85664 9.57488 6.54564 9.66362 7.29571C9.75237 8.04579 9.55491 8.80174 9.11069 9.41261"
+                stroke="#454D54"
+                stroke-linecap="round"
+                stroke-linejoin="round"
+              />
+              <path
+                d="M9.87855 3.71145L10.3786 3.21145C10.9412 2.64884 11.7042 2.33276 12.4999 2.33276C13.2955 2.33276 14.0586 2.64884 14.6212 3.21145C15.1838 3.77406 15.4999 4.53713 15.4999 5.33278C15.4999 6.12844 15.1838 6.8915 14.6212 7.45412L11.4392 10.6361C10.9056 11.1706 10.1901 11.4841 9.43546 11.5142C8.68083 11.5442 7.94263 11.2885 7.36826 10.7981C6.79389 10.3077 6.4256 9.61878 6.33692 8.86877C6.24825 8.11877 6.44571 7.36291 6.88988 6.75212"
+                stroke="#454D54"
+                stroke-linecap="round"
+                stroke-linejoin="round"
+              />
+            </svg>
```

Comment:
I just saw Aaron raising a [ticket regarding this](https://nofrixion.atlassian.net/browse/MOOV-1764). It seems we're having some trouble when we put SVGs directly into the code. We need to make sure we're capitalising the dashed properties.

Here's what I mean:

`stroke-linecap` should be `strokeLinecap`
`stroke-linejoin` should be `strokeLinejoin`

---

This is a peculiar React quirk, so we should make sure we're doing this consistently across all SVG files. Since you're on it, could you please make these changes in the SVGs that have this issue? I believe there are only a few.

This only applies for SVGs being used directly on the code, not when they're being used from the `<img/>` tag.

Reply:
Oh yes. I'm surprised that this sneaked in because I was getting error for the other svgs so I changed it. But didn't get it for this one. :(

Reply:
Done.